File: Core/src/main/kotlin/jokatu/game/event/dialog/DialogRequestBuilderNeedingInputType.java
package jokatu.game.event.dialog;

import jokatu.game.input.Input;
import jokatu.game.player.Player;
import org.jetbrains.annotations.NotNull;

public interface DialogRequestBuilderNeedingInputType<P extends Player> {

	@NotNull
	<I extends Input> DialogRequestBuilderNeedingConsumer<P, I> withInputType(@NotNull Class<I> inputType);
}

File: Core/src/main/kotlin/jokatu/game/event/dialog/DialogRequestBuilderNeedingInputType.kt
package jokatu.game.event.dialog

import jokatu.game.input.Input
import jokatu.game.player.Player
import kotlin.reflect.KClass

interface DialogRequestBuilderNeedingInputType<P : Player> {

	fun <I : Input> withInputType(inputType: KClass<I>): DialogRequestBuilderNeedingConsumer<P, I>
}
